### ISSUE TITLE
Portable Services vs. SELinux

### ISSUE THREAD

**Issue by keithy** [systemd/systemd#13930]
The question of SELinux support was raised a while ago https://github.com/systemd/systemd/pull/8620#issuecomment-391985960 so probably deserves an issue of its own to track progress.

In Fedora31 the public manifestation of this with a totally minimal service is
```
Failed to attach image: Image '/home/keith/goss-native' lacks os-release data, refusing.
```
the logs are
```
Nov 03 19:46:04 localhost.localdomain audit[19521]: AVC avc:  denied  { read } for  pid=19521 comm="systemd-portabl" name="os-release" dev="sda5" ino=15204900 scontext=system_u:system_r:init_t:s0 tcontext=unconfined_u:object_r:user_home_t:s0 tclass=file permissive=0
Nov 03 19:46:04 localhost.localdomain audit[19521]: AVC avc:  denied  { read } for  pid=19521 comm="systemd-portabl" name="goss-native.service" dev="sda5" ino=15204740 scontext=system_u:system_r:init_t:s0 tcontext=unconfined_u:object_r:user_home_t:s0 tclass=file permissive=0
```
So importing user supplied files needs a little more thought/documentation.
Copying into /var/lib/portables in order to pick up the label var_lib_t ought to fix this, but perhaps portablectl could explicitly fixup the labelling (is it allowed to?), or explain the problem.

Having fixed the above, the next 
https://github.com/coreos/fedora-coreos-tracker/issues/312#issue-516926078

**Comment by poettering** [comment]
Uh, I am not an selinux guy. We could probably specify contex= among the the mount options when we mount images, but quite frankly I don't know selinux well enough to do this, and I figure this also requires selinux policy changes which I don't grok.

**Comment by mathieu-lacage** [comment]
@keithy I might have misunderstood your suggestion but, I have attempted to copy my raw image to /var/libportables/ and I still get:

```
$ sudo portablectl attach pdns
(Matching unit files with prefix 'pdns'.)
AttachImage failed: Access denied
$
```

```
Apr 12 18:07:59 Host-005 audit[605046]: AVC avc:  denied  { read write } for  pid=605046 comm="systemd-portabl" name="loop-control" dev="devtmpfs" ino=215 scontext=system_u:system_r:init_t:s0 tcontext=system_u:object_r:loop_control_device_t:s0 tclass=chr_file permissive=0
```

Of course, disabling selinux makes this command work...

**Comment by karimfromjordan** [comment]
I just came across the same issue. Even if the portable image is labeled correctly using `sudo restorecon -Rv /var/lib/portables` attaching the image returns  a `loop-control` error. I haven't come across an actual solution. Some suggest to create a SELinux policy based on the error but that's super tedious and once one error is fixed the next comes up. The only solution is to disable SELinux which isn't recommended for security reasons.

**Comment by RulentWave** [comment]
> I just came across the same issue. Even if the portable image is labeled correctly using `sudo restorecon -Rv /var/lib/portables` attaching the image returns  a `loop-control` error. I haven't come across an actual solution. Some suggest to create a SELinux policy based on the error but that's super tedious and once one error is fixed the next comes up. The only solution is to disable SELinux which isn't recommended for security reasons.

Did ever find a workaround? I just came across this myself on fedora

**Comment by karimfromjordan** [comment]
> > I just came across the same issue. Even if the portable image is labeled correctly using `sudo restorecon -Rv /var/lib/portables` attaching the image returns  a `loop-control` error. I haven't come across an actual solution. Some suggest to create a SELinux policy based on the error but that's super tedious and once one error is fixed the next comes up. The only solution is to disable SELinux which isn't recommended for security reasons.
> 
> Did ever find a workaround? I just came across this myself on fedora

No. I don't see any other way currently than to simply avoid any linux distro that uses SELinux. That means for now I'm forced to use Ubuntu Server even though I'd rather use Fedora.
